aten/src/ATen/Context.cpp

```diff
@@ -105,11 +146,26 @@ void Context::setUserEnabledNNPACK(bool e) {
   enabled_nnpack = e;
 }
 
-bool Context::allowTF32CuDNN() const {
+bool Context::allowTF32CuDNN(const std::string& op) const {
+  if (op.size() == 0){
+    bool allow_tf32_rnn = float32Precision("cuda", "rnn") == "tf32";
+    bool allow_tf32_conv = float32Precision("cuda", "conv") == "tf32";
+    TORCH_CHECK(
+        allow_tf32_rnn == allow_tf32_conv && allow_tf32_rnn == allow_tf32_cudnn,
+        "You are checking whether allow_tf32 is enabled for cudnn backend without a specific operator name. ",
+        "But the current status indicate that cudnn conv and cudnn rnn have different TF32 status with cudnn",
+        "This indicate you have used mix of legacy API and new API to set the TF32 status. ",
+        "We suggest to only use the new API to set the TF32 status, you can check the usage at ",
+        "https://github.com/pytorch/pytorch/blob/main/docs/source/notes/cuda.rst#tensorfloat-32-tf32-on-ampere-and-later-devices");
```

Comment:
nit:
"PyTorch is checking whether allow_tf32 is enabled for cuDNN without a specific operator name, but the current flag(s) indicate that cuDNN conv and cuDNN RNN have different TF32 flags. This combination indicates that you have used a mix of the legacy and new APIs to set the TF32 flags. We suggest only using the new API to set the TF32 flag(s). See also:  https://github.com/pytorch/pytorch/blob/main/docs/source/notes/cuda.rst#tensorfloat-32-tf32-on-ampere-and-later-devices"

Reply:
Updated, thanks for your advice.